bool leader
bool all_together
int8[] robots